### PR TITLE
Added docs for GCE, DO and Choosing a Provider

### PR DESCRIPTION
**Pull request by arkigos**
## Description

We are adding GCE and DO to our supported providers. These are the Getting Started docs for those providers. Also, a doc comparing all of our providers. NOTE this does not update the navbar links and so these docs will not be linked to until we make that change (presumably after our providers go live).
## Motivation and Context

We needed docs for our new providers! (and a meta doc to compare them)
## How Has This Been Tested?

N/A (could use a few proofreads for typos?)
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by sinan** [comment]
thx @arkigos, @sibblegp can you please review the language and content.

File: docs/docs/choose-provider/index.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+ * ***[AWS Documentation](https://aws.amazon.com/vpc/)***
+ * ***[Using IAM in your Stack](/docs/create-an-aws-vpcr)***
+ * ***[Stack Reference](/docs/awc_vpc)***
+
+***
+
+## Vagrant <a name="vagrant"></a>
+
+Vagrant allows you to use your own machine to host your Koding VMs. You will install it along-side VirtualBox, so Vagrant can interface you with all of its features.
```
Comment (sibblegp) [inline docs/docs/choose-provider/index.md]: "You will install it" -> "It gets installed"

File: docs/docs/choose-provider/index.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+***
+
+## DigitalOcean <a name="digital-ocean"></a>
+
+***[Getting Started with DigitalOcean](/docs/getting-started-digital-ocean)***
+***[DigitalOcean Website](https://www.digitalocean.com/)***
+***[DigitalOcean Stack Reference](https://www.terraform.io/docs/providers/do/index.html)***
+
+DigitalOcean is a very popular option among developers an teams who don't need all of the features of AWS or GCE and also don't need all of the headache. Its feature-set is much smaller than the others, but those it does offer are the features most used by small teams. Documentation is limited (though this is countered by a strong active community), but since they are such a straightforward service, there isn't much required. DigitalOcean's pricing is among the best, in large-part due to their bare-bones approach.
```
Comment (sibblegp) [inline docs/docs/choose-provider/index.md]: "an" -> "and".  Remove the line about headache.  "smaller" -> "simpler".  Remove the word strong in "strong active community".

File: docs/docs/choose-provider/index.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+***[DigitalOcean Stack Reference](https://www.terraform.io/docs/providers/do/index.html)***
+
+DigitalOcean is a very popular option among developers an teams who don't need all of the features of AWS or GCE and also don't need all of the headache. Its feature-set is much smaller than the others, but those it does offer are the features most used by small teams. Documentation is limited (though this is countered by a strong active community), but since they are such a straightforward service, there isn't much required. DigitalOcean's pricing is among the best, in large-part due to their bare-bones approach. 
+
+> Using DigitalOcean as a provider will require you to [Sign Up](https://cloud.digitalocean.com/registrations/new) for the service. Currently, this requires a credit card and for you to pre-load your account with at least $5 (the amount needed for the lowest-tier droplet for a month.) Using DigitalOcean in your Stack will require you to create and copy an Access Token, [here](https://cloud.digitalocean.com/settings/api/tokens). This is gone over in greater detail in our [Getting Started with DigitalOcean](/docs/getting-started-digital-ocean) guide.
```
Comment (sibblegp) [inline docs/docs/choose-provider/index.md]: Clarify what a droplet is.

File: docs/docs/choose-provider/index.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+* **Floating IP**
+A Floating IP is a static public IP address that can dynamically map to any VM. This can be used to mask failures by re-routing domains to different instances without needing to change the DNS records. Configuring this is very simple (pick one of your droplets and click a button).
```
Comment (sibblegp) [inline docs/docs/choose-provider/index.md]: If we can configure floating IPs in Koding, then let's not mention selecting a droplet and clicking a button.

File: docs/docs/choose-provider/index.md [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+
+***
+
+## Google Compute Engine <a name="google-compute-cloud"></a>
+
+***[Getting Started with Google Compute Engine](/docs/getting-started-with-google-compute-engine)***
+***[Google Cloud Platform Website](https://cloud.google.com/compute/)***
+***[Google Compute Engine Stack Reference](https://www.terraform.io/docs/providers/google/index.html)***
+
+Google Compute Engine is packed with many features which rival those of AWS. GCE offers competitive pricing, an astonishing uptime, and is fast!
```
Comment (sibblegp) [inline docs/docs/choose-provider/index.md]: "competitive pricing, astonishing uptime, and speed!"

File: docs/docs/choose-provider/index.md [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+Like with most Cloud Providers, GCE offers a host of different OS images to build your VM with. 
+
+ * ***[GCE Images](https://cloud.google.com/compute/docs/images)***
+ * ***[Stack Reference](https://www.terraform.io/docs/providers/google/r/compute_image.html)***
+* **Global Address**
+A Global Address is a static public IP address that can dynamically map to any VM. This can be used to mask failures by re-routing domains to different instances without needing to change the DNS records. 
+
+ * ***[Configuring an Instance's Address](https://cloud.google.com/compute/docs/configure-instance-ip-addresses)***
+ * ***[Stack Reference](https://www.terraform.io/docs/providers/google/r/compute_global_address.html)***
+* **Access Control**
```
Comment (sibblegp) [inline docs/docs/choose-provider/index.md]: Not Koding related.

File: docs/docs/getting-started-digitalocean/index.md [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+
+Using DigitalOcean as your provider will require you to sign up on their website. It will also require you to enter your credit card information and make an initial payment of at least $5. You can do that [here](https://cloud.digitalocean.com/registrations/new). 
+
+[Learn more about pricing](https://www.digitalocean.com/pricing/) at DigitalOcean.
+
+***
+
+## Create your Personal Access Token <a name="create-pat"></a>
+
+Your DigitalOcean Personal Access Token will provide the credentials you need to instantiate your Stacks using DigitalOcean.
```
Comment (sibblegp) [inline docs/docs/getting-started-digitalocean/index.md]: "using DigitalOcean on Koding."

File: docs/docs/getting-started-digitalocean/index.md [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+***
+
+## Create your Stack <a name="create-stack"></a>
+
+To create your Stack:
+
+1. [Login to Koding](https://koding.com/Teams/Select)
+2. Go to your [Dashboard](https://relepic.koding.com/Home) OR click [Stacks](https://relepic.koding.com/Home/Stacks) from your sidebar.
+3. Either way, you'll default to the Stacks screen, so from that screen click [New Stack](https://relepic.koding.com/Stack-Editor/New)
+4. Use the Wizard to create and  populate your Stack with some defaults, make sure you select Digital Ocean as provider
```
Comment (sibblegp) [inline docs/docs/getting-started-digitalocean/index.md]: Make selecting Digital Ocean a specific step and then say they can optionally skip the remaining steps or continue to install some default software.

File: docs/docs/getting-started-digitalocean/index.md [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+***
+
+## Set your Credentials <a name="set-creds"></a>
+
+To add your DigitalOcean credentials:
+
+1. From your Edit Stack screen (if you've just created the Stack, you will be on that screen by default) click on the Credentials tab
+2. On the bottom right of the Credentials screen, click Create New
+3. Name your Credentials whatever you like (pick something that will help you recognize them at a glance)
+4. Copy the Personal Access token that you generated [earlier](#create-pat)
```
Comment (sibblegp) [inline docs/docs/getting-started-digitalocean/index.md]: "Copy"  ->  "Paste in"

File: docs/docs/getting-started-digitalocean/index.md [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+5. Click Save
+6. You'll be taken back the Credentials tab where you'll see your new credentials listed by name. Click Use This & Continue to set them as the credentials for this stack. 
+
+***
+
+## Save your Stack and Initialize <a name="save-init"></a>
+
+Once you've created your Stack and set our credentials:
```
Comment (sibblegp) [inline docs/docs/getting-started-digitalocean/index.md]: "set your credentials"

File: docs/docs/getting-started-google-compute-enging/index.md [hunk 11]
```diff
@@ -0,0 +1,137 @@
+# Getting Started with Google Compute Engine
+
+Getting started with GCE requires a few
```
Comment (sibblegp) [inline docs/docs/getting-started-google-compute-enging/index.md]: Sentence fragment.

File: docs/docs/getting-started-google-compute-enging/index.md [hunk 12]
```diff
@@ -0,0 +1,137 @@
...
+
+Google Cloud Platform starts you out in a default 'project' called My First Project. Projects are just a way of organizing your resources to be associated with the project they are used for. 
+
+You might not want to stick with this default project, so go ahead and create a new one:
+
+1. From the Dashboard or Home screen, you'll see a drop-down at the top of the screen with the name of your project displayed (defaulted to My First Project). Click that to open the menu.
+2. Click Create Project.
+3. This will open a dialog screen for you to name your project. Click Show Advanced Options to select your Region. This is important even if you don't change the region, because you'll be asked for the Region later when setting up credentials for your Stack.
+
+> Take note not only of the Project Name but the Project ID. You will need this ID when creating credentials for your Koding Stack.
```
Comment (sibblegp) [inline docs/docs/getting-started-google-compute-enging/index.md]: I would emphasize that they should write down or copy the project ID somewhere.  Taking note of it doesn't mean they'll remember it.

File: docs/docs/getting-started-google-compute-enging/index.md [hunk 13]
```diff
@@ -0,0 +1,137 @@
...
+
+
+***
+
+## Create your Stack <a name="create-stack"></a>
+
+To create your Stack:
+
+1. [Login to Koding](https://koding.com/Teams/Select)
+2. Go to your [Dashboard](https://relepic.koding.com/Home) OR click [Stacks](https://relepic.koding.com/Home/Stacks) from your sidebar.
```
Comment (sibblegp) [inline docs/docs/getting-started-google-compute-enging/index.md]: These links are to a specific team.  You can't link to a person's dashboard directly from the docs.  Make sure this is also the case with other dashboard links.

File: docs/docs/getting-started-google-compute-enging/index.md [hunk 14]
```diff
@@ -0,0 +1,137 @@
...
+3. Either way, you'll default to the Stacks screen, so from that screen click [New Stack.](https://relepic.koding.com/Stack-Editor/New)
```
Comment (sibblegp) [inline docs/docs/getting-started-google-compute-enging/index.md]: Same problem of linking to a specific team.

File: docs/docs/getting-started-google-compute-enging/index.md [hunk 15]
```diff
@@ -0,0 +1,137 @@
...
+4. Use the Wizard to create and  populate your Stack with some defaults, make sure you select Google Compute Engine as provider.
```
Comment (sibblegp) [inline docs/docs/getting-started-google-compute-enging/index.md]: Make selecting GCE as a specific step.

File: docs/docs/getting-started-google-compute-enging/index.md [hunk 16]
```diff
@@ -0,0 +1,137 @@
...
+
+Make whatever changes you'd like to this script. Our [Stack Reference for Google](https://www.terraform.io/docs/providers/google/index.html) will help you see what is possible.
+
+> You won't need to worry about the google_* variables. These will be automatically populated when you set up credentials in the next step. 
+
+***
+
+## Set your Credentials <a name="set-creds"></a>
+
+To add your DigitalOcean credentials:
```
Comment (sibblegp) [inline docs/docs/getting-started-google-compute-enging/index.md]: GCE credentials

File: docs/docs/getting-started-google-compute-enging/index.md [hunk 17]
```diff
@@ -0,0 +1,137 @@
...
+
+1. From your Edit Stack screen (if you've just created the Stack, you will be on that screen by default) click on the Credentials tab
+2. On the bottom right of the Credentials screen, click Create New.
+3. Name your Credentials whatever you like (pick something that will help you recognize them at a glance)
+4. Enter your Project ID. If you don't know it, you should be able to find it [here](https://console.cloud.google.com/iam-admin/projects). 
+5. Copy the contents of the Service Account Key file you saved to your computer [earlier.](#create-key)
+6. Set the region of your project.
```
Comment (sibblegp) [inline docs/docs/getting-started-google-compute-enging/index.md]: "Make sure this is the same region you set when creating your project"

**Review by sibblegp — CHANGES_REQUESTED** [review]
Typos and other changes.

**Comment by gokmen** [comment]
@arkigos can u take a look to issues commented by @sibblegp

**Comment by gokmen** [comment]
ping.

**Comment by sinan** [comment]
since these aren't referenced anywhere I merged them, @kwagdy is going to fix and push again.

**Pull request merged**